blueman/plugins/applet/DBusService.py

```diff
@@ -17,20 +16,37 @@
 from blueman.services.meta import SerialService, NetworkService
 
 
+class RFCOMMConnectedListener:
+    def on_rfcomm_connected(self, service: Service, port: str) -> None:
+        ...
+
+    def on_rfcomm_disconnect(self, port: int) -> None:
+        ...
+
+
+class RFCOMMConnectHandler:
+    def rfcomm_connect_handler(self, service: Service, reply: Callable[[str], None],
+                               err: Callable[[Exception], None]) -> bool:
+        ...
+
+
+class ServiceConnectHandler:
+    def service_connect_handler(self, service: Service, ok: Callable[[], None],
+                                err: Callable[[Union[NMConnectionError, GLib.Error]], None]) -> bool:
```

Comment:
I just wanted to make this explicit so that a caller can immediately get what the error callback is expected to get exactly.

At runtime, both are `Exception`s, yes. `GLib.Error` (an alias for `GLib.GError`) is actually even a `RuntimeError`. For mypy, unfortunately, `GLib.Error` is currently `Any`, just like all the `gi.repository` types, as we do not have stubs in place for them, so that it even prevents proper type checks. I'm planning on using something like https://github.com/pygobject/pygobject-stubs but will have to improve it first as GObject stubs are all far from correct and complete.

I definitely like to have those types explicit,  however, `Exception` would even be better for static analysis as `Union[NMConnectionError, GLib.Error]` is currently just `Any`. If you like that better, I'm fine with changing all `Union`s I add here to just `Exception` or `Union[Exception, str]` as they all include `GLib.Error`.

Reply:
Don't use the stubs please. They are pretty much unmaintained. There is an effort to generate them from the introspection data. I can dig up the pr if you want it.

Reply:
https://github.com/pygobject/pgi-docgen/pull/176